mitodl/open-discussions
# Calculate and return 'is_favorite' and 'lists' fields in ES search results

#### Pre-Flight checklist

#### What are the relevant tickets?
Closes #2422 

#### What's this PR do?
- Uses scripted fields to include user-specific "is_favorite" and "lists" fields in each ES search result
- Removes API calls to retrieve all favorites and lists on the `CourseSearchPage`

#### How should this be manually tested?
- Run a search on the course search page.  The stars should be filled on any items that you've already favorited or added to lists.
- Add more search results as favorites or to lists, and unfavorite/unlist others.  The stars for the results should change appropriately.
